### ISSUE TITLE
`delete-pipeline` returns strange error but succeeds

### ISSUE THREAD

**Issue by derekchiang** [pachyderm/pachyderm#281]
```
$  pachctl delete-pipeline wordcount-map
2016/04/12 15:45:09 transport: http2Client.notifyError got notified that the client transport was broken EOF.
2016/04/12 15:45:10 transport: http2Client.notifyError got notified that the client transport was broken EOF.
2016/04/12 15:45:10 transport: http2Client.notifyError got notified that the client transport was broken EOF.
2016/04/12 15:45:11 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 104.154.29.73:30650: getsockopt: connection refused"; R
econnecting to "104.154.29.73:30650"
2016/04/12 15:45:11 transport: http2Client.notifyError got notified that the client transport was broken read tcp 10.238.184.15:51646->104.154.29.73:30650: read: connection reset by 
peer.
```

But the pipeline in fact gets deleted.

**Comment by sjezewski** [comment]
Hmmm I've seen that pattern in several places before. I _think_ I see it most when the server errors. If it recovers, those errors will go away once the client reconnects. Otherwise, if its thrashing, you'll continue to see them.

**Comment by jdoliner** [comment]
Yeah most likely cause of this is that the `DeletePipeline` request caused the server to crash, which broke the connection.

**Comment by derekchiang** [comment]
Fixed in #297
